docs/hosting/composer.md

```diff
@@ -0,0 +1,238 @@
+only include the core farmOS modules.
+
+If you want to include additional modules that are created by yourself or
+others in the community, you may find that you need more control over the
+build process. This guide outlines the basic workflow for managing your
+farmOS-based projects with Composer.
+
+## Composer projects
+
+First, it's important to understand the concept of a "Composer project".
+
+If you are building a new PHP application from scratch with Composer, the
+first step is to create a `composer.json` file. As described in the Composer
+docs:
+
+> To start using Composer in your project, all you need is a `composer.json`
+> file. This file describes the dependencies of your project and may contain
+> other metadata as well.
+
+https://getcomposer.org/doc/01-basic-usage.md#composer-json-project-setup
+
+### Project templates
+
+Composer also provides a `create-project` command for starting a new project
+from an existing one. This allows for "project templates" to be provided for
+developers to build on top of.
+
+> You can use Composer to create new projects from an existing package. This
+> is the equivalent of doing a git clone/svn checkout followed by a
+> `composer install` of the vendors.
+
+https://getcomposer.org/doc/03-cli.md#create-project
+
+A good example of this is Drupal's `recommended-project` template, which is
+described as follows:
+
+> These project templates serve as a starting point for creating a
+> Composer-managed Drupal site. Once you have selected a template project and
+> created your own site, you will take ownership of your new project files.
+> Thereafter, future updates will be done with Composer.
+
+https://www.drupal.org/docs/develop/using-composer/starting-a-site-using-drupal-composer-project-templates
+
+For example, to create a new project based on Drupal's `recommended-project`
+template:
+
+    composer create-project drupal/recommended-project
+
+This will initialize a `composer.json` and `composer.lock` file that serves as
+a template for a Drupal website.
+
+### farmOS project template
+
+farmOS provides its own project template:
+https://packagist.org/packages/farmOS/project
+
+The official farmOS releases are built from this template. It can also be used
+to start your own project based on farmOS. This process is described in the
+next section.
+
+## Starting your farmOS project
+
+To start your own project based on the farmOS project template, open a new
+directory and run the following command:
+
+    composer create-project farmos/project
+
+This will create a `composer.json` file in your directory, copied directly from
+the [farmOS project template](https://packagist.org/packages/farmos/project).
+
+You can use this `composer.json` as the starting point for your own project's
+code repository with [Git](https://git-scm.com/):
+
+    git init
+    git add composer.json
+    git commit -m 'Intitial commit of my farmOS project.'
```

Comment:
I think we'll get better (more consistent) syntax highlighting with "fences" and a type hint...

~~~markdown
```sh
git init
git add composer.json
git commit -m 'Intitial commit of my farmOS project.'
```
~~~

Reply:
Good idea.

Changed here: https://github.com/farmOS/farmOS/pull/648/commits/49b3b9a65ed3d04e005f918488780f9dfd88de1e